src/mpid/ch4/include/mpidpre.h

```diff
@@ -184,7 +200,11 @@ typedef struct MPIDIG_req_ext_t {
         MPIDIG_acc_req_t areq;
     };
 
-    MPIDIG_rreq_async_t async;
+    int handler_id;
+    union {
+        MPIDIG_rreq_async_t async;
+        MPIDIG_sreq_async_t send_async;
```

Comment:
It'll be nice to rename `async` to `recv_async` -- maybe in a separate commit.

Reply:
ok